stdlib/public/core/FloatingPointParsing.swift.gyb

```diff
@@ -78,32 +77,58 @@ extension ${Self}: LosslessStringConvertible {
   ///       let e = ${Self}("2837.5e-2")
   ///       // e == 28.375
   ///
-  /// - A *hexadecimal value* contains the significand, either `0X` or `0x`,
-  ///   followed by a sequence of hexadecimal digits. The significand may
+  ///   A decimal string is converted into a correctly-rounded ${Self}
+  ///   value using IEEE 754 round-to-nearest
+  ///   conventions. Very small values are converted to positive or
+  ///   negative zero, very large values are converted
+  ///   to plus or minus infinity.
+  ///
+  ///       let y = ${Self}("1.23e-9999")
+  ///       // y == 0.0
+  ///       // y?.sign == .plus
+  ///
+  ///       let z = ${Self}("-7.89e-7206")
+  ///       // z == -0.0
+  ///       // z?.sign == .minus
+  ///
+  ///       let r = ${Self}("1.23e17802")
+  ///       // r == ${Self}.infinity
+  ///
+  ///       let s = ${Self}("-7.89e7206")
+  ///       // s == ${Self}.-infinity
+  ///
+  /// - A *hexadecimal value* contains a significand consisting of
```

Comment:
It's a little bit weird that we call these decimal and hexadecimal _values_ instead of "strings" or "character sequences" or similar. I'm not entirely sure what I want to do about it, however. I think I would prefer to use "value" only for the numeric value and refer to the string as either "string" or "character sequence".

Reply:
Nice point.  I'll take another sweep and see if I can tighten that up a bit more.

Reply:
I made a few more tweaks to be consistent about:
* "String" or "input" to refer to the input text
* "Number" or "value" to refer to the abstract Real
* "instance" to refer to the resulting ${Self}
I also tried to use the term "rounding" or "rounded" when the text is explicitly referring to differences between the Real value and the final ${Self} instance.